packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveSourcePath.ts

```diff
@@ -95,11 +99,59 @@ export async function forceSourceRetrieveSourcePath(explorerPath: vscode.Uri) {
       return;
     }
   }
+
   const commandlet = new SfdxCommandlet(
     new SfdxWorkspaceChecker(),
     new FilePathGatherer(explorerPath),
-    new ForceSourceRetrieveSourcePathExecutor(),
+    useBetaRetrieve(explorerPath)
+      ? new LibraryRetrieveSourcePathExecutor()
+      : new ForceSourceRetrieveSourcePathExecutor(),
     new SourcePathChecker()
   );
   await commandlet.run();
 }
+
+// this supported types logic is temporary until we have a way of generating the metadata type from the path
+// once we have the metadata type we can check to see if it is a toolingsupportedtype from that util
+
+function useBetaRetrieve(explorerPath: vscode.Uri): boolean {
```

Comment:
Scoping the retrieve exclusively to Apex Class for now.

Reply:
We can expose a method on deploy-retrieve that returns if a type is supported based on filePath, so here we don't even care about figuring out the metadatype.

Reply:
That's a good idea, I'll add it to my retrieve changes.

Reply:
That's a good idea, I'll add it to my retrieve changes.